public/Aryan Suri/Licenses/licenses.js

```diff
@@ -0,0 +1,255 @@
+							title: "CC BY",
+							link: "https://creativecommons.org/licenses/by/4.0/"
+						};
+					case "ccbysa":
+						return {
+							label: "cc-by-sa",
+							title: "CC BY-SA",
+							link: "https://creativecommons.org/licenses/by-sa/4.0/"
+						};
+					case "ccbyncsa":
+						return {
+							label: "cc-by-nc-sa",
+							title: "CC BY-NC-SA",
+							link: "https://creativecommons.org/licenses/by-nc-sa/4.0/"
+						};
+					case "ccbync":
+						return {
+							label: "cc-by-nc",
+							title: "CC BY-NC",
+							link: "https://creativecommons.org/licenses/by-nc/4.0/"
+						};
+					case "ccbynd":
+						return {
+							label: "cc-by-nd",
+							title: "CC BY-ND",
+							link: "https://creativecommons.org/licenses/by-nd/4.0/"
+						};
+					case "ccbyncnd":
+						return {
+							label: "cc-by-nc-nd",
+							title: "CC BY-NC-ND",
+							link: "https://creativecommons.org/licenses/by-nc-nd/4.0/"
+						};
+					case "gnu":
+						return {
+							label: "gnu",
+							title: "GNU GPL",
+							link: "https://www.gnu.org/licenses/gpl-3.0.en.html"
+						};
+					case "gnudsl":
+						return { label: "gnudsl", title: "GNU DSL", link: "https://www.gnu.org/licenses/dsl.html" };
+					case "gnufdl":
+						return {
+							label: "gnufdl",
+							title: "GNU FDL",
+							link: "https://www.gnu.org/licenses/fdl-1.3.en.html"
+						};
+					case "arr":
+						return { label: "arr", title: "All Rights Reserved Â©", };
+				}
+			}
+		}
+	}
+	return null; //not found
+}
+
+(function () {
+
+	$('body').append(`
+                <div id="warningModal">
+					<div id="warningModalContent">  
+                    </div>
+                </div>
+		`);
+
+	function controlCopy() {
+
+		const cc = getCC();
+		const admin = document.getElementById('adminHolder').innerText === 'true';
+		const pro = document.getElementById("proHolder").innerText === 'true';
+		const groups = document.getElementById('groupHolder').innerText.toLowerCase();
+
+		if (admin) { console.log("admin") }
+		else if (pro && groups.includes('developer')) { console.log("developer") }
+		else if (pro && (groups.includes('basicuser') || groups.includes('workshop'))) {
+			if (cc) {
+				localStorage.setItem("cc", cc.label);
+				console.log("basicuser");
+				switch (cc.label) {
+					case "arr":
+						$('body').bind('copy paste cut', function (e) {
+							e.preventDefault();
+							ccDetector();
+						});
+						break;
+					default: document.addEventListener("copy", ccDetector);
+				}
+			}
+			else {
+				document.addEventListener("copy", ccDetector);
+			}
+		} else {
+			if (cc) {
+				localStorage.setItem("cc", cc.label);
+				switch (cc.label) {
+					case "arr":
+						$('body').bind('copy paste cut', function (e) {
+							e.preventDefault();
+							ccDetector();
+						});
+						break;
+					default: document.addEventListener("copy", ccDetector);
+				}
+			}
+			else {
+				document.addEventListener("copy", ccDetector);
+			}
+		}
```

Comment:
These two statement blocks look identical. Is this on purpose? Maybe you meant for this block to be different such that "nonpro users can't copy NC content"

Reply:
yes this is on purpose there is a difference between pro basic user and non pro basic user

Reply:
Is it correct that currently the only functional difference is the console.log statement?
Is it correct that you have this set up like this so that you can differentiate them more in the future? I can't currently think of anything that would be different between these two, but you/Delmar might have something in mind.